corehq/motech/views.py

```diff
@@ -81,8 +83,16 @@ def get_queryset(self):
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
+        result = lookup_doc_id(context['log'].payload_id)
+        has_permission = False
+        if result:
+            doc_info = get_doc_info(result.doc)
+            if doc_info:
+                user = self.request.couch_user
+                has_permission = doc_info.user_has_permissions(context['log'].domain, user, result.doc)
+
         context.update({
-            "is_dimagi": self.request.couch_user.is_dimagi
+            "has_permission": has_permission,
```

Comment:
since you're already doing the doc lookup here you could just pass the link directly to the template rather than using the quick find view (which will do the lookup again).

```
link = None
if doc_info and doc_info.link and doc_info.user_has_permission(...):
    link = doc_info.link
```